Tests/SonarQube.Common.UnitTests/UtilitiesTests.cs

```diff
@@ -130,6 +133,191 @@ public void EnsureDirectoryExists_WhenDirectoryIsWhitespaces_ThrowsArgumentNullE
             action.ShouldThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("directory");
         }
 
+        [TestMethod]
+        public void EnsureDirectoryExists_WhenDirectoryMissing_IsCreated()
+        {
+            // Arrange
+            var baseDir =TestUtils.CreateTestSpecificFolder(this.TestContext);
+            string newDir = Path.Combine(baseDir, "newDir");
+            var logger = new TestLogger();
+
+            // Act
+            Utilities.EnsureDirectoryExists(newDir, logger);
+
+            // Assert
+            Directory.Exists(newDir).Should().BeTrue();
+            logger.Warnings.Count.Should().Be(0);
+            logger.Errors.Count.Should().Be(0);
+        }
+
+        [TestMethod]
+        public void EnsureDirectoryExists_WhenDirectoryExists_IsNoOp()
+        {
+            // Arrange
+            var baseDir = TestUtils.CreateTestSpecificFolder(this.TestContext);
+            var logger = new TestLogger();
+
+            // Act
+            Utilities.EnsureDirectoryExists(baseDir, logger);
+
+            // Assert
+            Directory.Exists(baseDir).Should().BeTrue();
+            logger.Warnings.Count.Should().Be(0);
+            logger.Errors.Count.Should().Be(0);
+        }
+
+        [TestMethod]
+        public void EnsureEmptyDirectory_WhenDirectoryIsInvalid_ThrowsArgumentNullException()
+        {
+            // 1. Null
+            Action action = () => Utilities.EnsureEmptyDirectory(null, new TestLogger());
+            action.ShouldThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("directory");
+
+            // 2. Empty
+            action = () => Utilities.EnsureDirectoryExists("", new TestLogger());
+            action.ShouldThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("directory");
+
+            // 3. Whitespace
+            action = () => Utilities.EnsureDirectoryExists("   ", new TestLogger());
+            action.ShouldThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("directory");
+        }
```

Comment:
The change in the product code to use _nameof_ resulting in those product lines being treated as uncovered.
I've added these invalid-arg tests. I've covered all three invalid cases in a single test (null, empty, whitespace) - these are pretty-low value tests, and having individual test methods for each case just adds noise and clutter.